src/extension/useExportControlToExtension.ts

```diff
@@ -0,0 +1,96 @@
+import _ from 'lodash';
+import { nanoid } from 'nanoid';
+import { useEffect, useState } from 'react';
+import { Control, useFormState, useWatch } from 'react-hook-form';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import { MessageData, UpdatePayload } from './types';
+import { proxyToObject } from './utils';
+
+const id = nanoid();
+
+export function useExportControlToExtension(control: Control<any>) {
+  const nestedFormValues = useWatch({ control });
+  const formState = useFormState({ control });
+
+  const [isExtensionEnabled, setIsExtensionEnabled] = useState(false);
+
+  const handleInitMessage = (message: MessageEvent<MessageData>) => {
+    if (
+      message.data.source !== 'react-hook-form-bridge' ||
+      message.data.type !== 'INIT'
+    ) {
+      return;
+    }
+    window.postMessage({
+      source: 'react-hook-form-bridge',
+      type: 'WELCOME',
+    } as MessageData);
+    setIsExtensionEnabled(true);
+  };
+
+  useEffect(() => {
+    window.addEventListener('message', handleInitMessage);
+    return () => window.removeEventListener('message', handleInitMessage);
+  }, []);
+
+  const toFlat = <V>(obj: object, defaultValue?: V) => {
```

Comment:
Could we move this into utility?

Reply:
It's not utility, needs `control._names`.
But sure, I can make it as untility

Reply:
Got you, you can make _names as payload? no biggy, just try to move some of the mapping logic into own function.